xgi/linalg/matrix.py

```diff
@@ -125,13 +128,16 @@ def adjacency_matrix(H, order=None, s=1, weighted=False, index=False):
         >>> ps = [0.01, 0.001]
         >>> H = xgi.random_hypergraph(n, ps)
         >>> A = xgi.adjacency_matrix(H)
-    """
 
+    """
     if index:
         I, rowdict, _ = incidence_matrix(H, index=True, order=order)
     else:
         I = incidence_matrix(H, index=False, order=order)
 
+    if I.shape == (0,):
+        return (np.array([]), {}) if index else np.array([])
```

Comment:
This works if we have no nodes, but when we have nodes but no edges (of order d), we want to return np.zeros((N,N)).

```
if I.shape == (0,):
    if not rowdict:
        A = np.array([])
    if not coldict:
        A = np.zeros((H.num_nodes, H.num_nodes))
    return (A, {}) if index else A 

```

Reply:
Good catch!